packages/atlas/src/views/viewer/VideoView/CommentThread.tsx

```diff
@@ -0,0 +1,263 @@
+
+import { CommentFieldsFragment } from '@/api/queries'
+import { Comment, CommentProps } from '@/components/_comments/Comment'
+import { CommentInput } from '@/components/_comments/CommentInput'
+import { ReactionId } from '@/config/reactions'
+import { absoluteRoutes } from '@/config/routes'
+import { useReactionTransactions } from '@/hooks/useReactionTransactions'
+import { useMemberAvatar } from '@/providers/assets'
+import { useConfirmationModal } from '@/providers/confirmationModal'
+import { useUser } from '@/providers/user'
+
+import { getCommentReactions } from './utils'
+
+type CommentThreadProps = {
+  onOpenSignInDialog: () => void
+  commentId: string
+  videoId?: string
+  videoAuthorId?: string
+  handleCommentReaction: (commentId: string, reactionId: ReactionId) => void
+  authorized: boolean
+  processingCommentReactionId: string | null
+  replies: (CommentFieldsFragment & { userReactions?: number[] })[] | null
+  channelOwnerMember?: string
+  isEditingCommentCollection: Set<string>
+  commentInputTextCollection: Map<string, string>
+  idx: number
+  onEditLabelClick: (replyComment?: CommentFieldsFragment) => void
+  onUpdateComment: ({ commentId }: { commentId: string }) => void
+  onEditCommentCancel: (comment: CommentFieldsFragment) => void
+  onSetIsEditingComment: ({ commentId, value }: { commentId: string; value: boolean }) => void
+  onSetCommentInputText: ({ commentId, comment }: { commentId: string; comment: string | undefined }) => void
+  onReplyDeleteClick: (replyComment: CommentFieldsFragment) => void
+} & CommentProps
+
+export const CommentThread: React.FC<CommentThreadProps> = ({
+  onOpenSignInDialog,
+  commentId,
+  videoId,
+  videoAuthorId,
+  handleCommentReaction,
+  authorized,
+  processingCommentReactionId,
+  reactionPopoverDismissed,
+  replies,
+  channelOwnerMember,
+  isEditingCommentCollection,
+  commentInputTextCollection,
+  idx,
+  onEditLabelClick,
+  onUpdateComment,
+  onEditCommentCancel,
+  onSetIsEditingComment,
+  onSetCommentInputText,
+  onReplyDeleteClick,
+  ...commentProps
+}) => {
+  const [repliesOpen, setRepliesOpen] = useState(false)
+  const [replyInputOpen, setReplyInputOpen] = useState(false)
+  const [highlightedComment, setHighlightedComment] = useState<string | null>(null)
+  const { activeMemberId, activeMembership } = useUser()
+  const { isLoadingAsset: isMemberAvatarLoading, url: memberAvatarUrl } = useMemberAvatar(activeMembership)
+  const { addComment, commentInputIsProcessingCollection } = useReactionTransactions()
+  const commentInputRef = useRef<HTMLTextAreaElement>(null)
+  const [openCancelConfirmationModal, closeCancelConfirmationModal] = useConfirmationModal()
+  const placeholderItems = !replies ? Array.from({ length: 4 }, () => ({ id: undefined })) : []
+  const replyAvatars = replies?.map((comment) => ({
+    url: comment?.author.metadata.avatar?.__typename === 'AvatarUri' ? comment?.author.metadata.avatar?.avatarUri : '',
```

Comment:
`useMemberAvatar` should be used for this, we need to get rid of this pattern

Reply:
This will be resolved here https://github.com/Joystream/atlas/issues/2735